common/views/components/IIIFImagePreview/ImageViewerImage.js

```diff
@@ -0,0 +1,77 @@
+// @flow
+import fetch from 'isomorphic-unfetch';
+import openseadragon from 'openseadragon';
+import { useState, useEffect } from 'react';
+import { spacing } from '../../../utils/classnames';
+
+function setupViewer(imageInfoSrc, viewerId, handleScriptError) {
+  fetch(imageInfoSrc)
+    .then(response => response.json())
+    .then(response => {
+      openseadragon({
+        id: `image-viewer-${viewerId}`,
+        visibilityRatio: 1,
+        showFullPageControl: false,
+        showHomeControl: false,
+        zoomInButton: `zoom-in-${viewerId}`,
+        zoomOutButton: `zoom-out-${viewerId}`,
+        showNavigator: true,
+        controlsFadeDelay: 0,
+        animationTime: 0.5,
+        tileSources: [
+          {
+            '@context': 'http://iiif.io/api/image/2/context.json',
+            '@id': response['@id'],
+            height: response.height,
+            width: response.width,
+            profile: ['http://iiif.io/api/image/2/level2.json'],
+            protocol: 'http://iiif.io/api/image',
+            tiles: [
+              {
+                scaleFactors: [1, 2, 4, 8, 16, 32],
+                width: 400,
+              },
+            ],
+          },
+        ],
+      });
+    })
+    .catch(_ => {
+      handleScriptError();
+    });
+}
+
+const ErrorMessage = () => (
+  <div
+    className={`image-viewer__error ${spacing(
+      { s: 5 },
+      { padding: ['left', 'right', 'top', 'bottom'] }
+    )}`}
+  >
+    <p>The image viewer is not working.</p>
+  </div>
+);
+
+type Props = {|
+  id: string,
+  infoUrl: string,
+|};
+
+const ImageViewerImage = ({ id, infoUrl }: Props) => {
+  const [scriptError, setScriptError] = useState(false);
+  const handleScriptError = () => {
+    setScriptError(true);
```

Comment:
Would it make sense to have 3 states, loading, loaded, and error?
Otherwise we might get a flash of the error first?

The `DoubleL` could be good?

Reply:
Would it be worth sending any errors along to Sentry here, too?

Reply:
Do you guys mind if we get this in, without this for testing on Thurs and I'll add it to my todo list?